### DISCUSSION TITLE
Node.js, Tiptap and React.renderToStaticMarkup : Empty result

### DISCUSSION THREAD

**Discussion by Taelar** [ueberdosis/tiptap#5326] [General Questions] [answered]
Greetings, 

I'm facing a rather complex case where I would like to render a tiptap editor server side, using `ReactDOM.renderToStaticMarkup`.

Basically, I have a tiptap editor (with some custom plugins) on front-end for users to create documents. Once done, users can render (print/download) their documents (with some dynamic layout around it) as pdf. The transformation to pdf is done server side with node.js. 
To achieve that, I would like to render a tiptap editor through `ReactDOM.renderToStaticMarkup`, and use this html as body content for the PDF generation (something we do with other types of content).

My problem is that once passed through the renderToStaticMarkup function, the editor content does not appear at all. You can find a minimal reproduction setup here : [CodeSandBox](https://codesandbox.io/p/devbox/beautiful-flower-rw5c5t).
My guess is that renderToStaticMarkup must only do one render while the EditorProvider component must need more to inject its content in the DOM.

While I understand that Tiptap is probably designed as a mainly front-end tool, I'd like to know if there is a way to achieve this kind of rendering server-side.

Thanks in advance
(I'm rather new to Github discussions and good practices around it, please let me know if I missed a step while creating this post)

**Comment by nperez0111** [comment, marked as answer]
There is a package '@tiptap/html' that could take editor content as json and render it to an html string. This would be the recommended way to transform the content. You could even embed its result into a react page with dangerously set inner html. 

The editor heavily relies on DOM APIs for rendering so rendering it on the server is just not an option unless you essentially want a whole chrome instance like puppeteer. 

It depends on your use case

**Reply by Taelar** [reply]
Thank you for your answer.
Indeed, I tried some things with puppeteer, but nothing really worked for me (maybe because I don't know a lot about puppeteer). Before spending more energy into that, I think I'll try something less "brutal" by injecting only CSS to render the PDF.
